lib/service_pattern/service.rb

```diff
@@ -6,7 +6,7 @@ def self.chain(*args, &blk)
     can_execute_response = service.can_execute?
     ServicePattern::Service.fail!(can_execute_response.errors) unless can_execute_response.success?
 
-    service.execute
+    service.perform
```

Comment:
So the idea was to continue to call `execute` but `execute` would be an in-between method that would catch the fail errors from perform.

Reply:
But this should be perform since execute will now catch the errors which contradicts the comment? :-)